pkg/resources/factory.go

```diff
@@ -78,32 +78,44 @@ func (rf *resourceFactory) GetSelector(attr string, values []string) (types.Devi
 
 // GetResourcePool returns an instance of resourcePool
 func (rf *resourceFactory) GetResourcePool(rc *types.ResourceConfig, deviceList []types.PciNetDevice) (types.ResourcePool, error) {
-	filteredDevice := deviceList
-
-	// filter by vendor list
-	if rc.Selectors.Vendors != nil && len(rc.Selectors.Vendors) > 0 {
-		if selector, err := rf.GetSelector("vendors", rc.Selectors.Vendors); err == nil {
-			filteredDevice = selector.Filter(filteredDevice)
+	filteredDevices := make([]types.PciNetDevice, 0)
+	filteredDevicesSet := make(map[types.PciNetDevice]bool)
+	for _, selector := range rc.Selectors {
```

Comment:
Just a thought on the implementation of selector loop, what about first collecting vendor/device/driver lists from each selector, then loop against the filters once. for example:
```
	venList := make([]string, 0)
	devList := make([]string, 0)
	drvList := make([]string, 0)

	filteredDevice := deviceList

	for _, selector := range rc.Selectors {
		venList = append(venList, selector.Vendors...)
		devList = append(devList, selector.Devices...)
		drvList = append(drvList, selector.Drivers...)
	}

	glog.Infof("selector vendor list: %v", venList)
	glog.Infof("selector device list: %v", devList)
	glog.Infof("selector driver list: %v", drvList)

	if venList != nil && len(venList) > 0 {
		if s, err := rf.GetSelector("vendors", venList); err == nil {
			filteredDevice = s.Filter(filteredDevice)
		}
	}

	if devList != nil && len(devList) > 0 {
		if s, err := rf.GetSelector("devices", devList); err == nil {
			filteredDevice = s.Filter(filteredDevice)
		}
	}

	if drvList != nil && len(drvList) > 0 {
		if s, err := rf.GetSelector("drivers", drvList); err == nil {
			filteredDevice = s.Filter(filteredDevice)
		}
	}

	glog.Infof("filtered devices : %v", filteredDevice)

```

Reply:
if use this scenario:
```
{
    "resourceList":
    [
        {
            "resourceName": "sriov_net_B",
            "selectors":[
                    {
                        "vendors": ["A"],
                        "devices": ["d1"]
                    },
            {
                        "vendors": ["B"],
                        "ifnames": ["d2"]
                    }
                ]
        }
    ]
}
```
Then devices d1 and d2 will be allowed for both vendor **A** and **B**, but the wanted result is only d1 from A and d2 from B

Reply:
Thanks for the example! 
I think it might be worth a note or explanation in the README on which devices will be selected with each example configuration.